### PR TITLE
Change usage of DPC++ to SYCL in DirectProgramming/DPC++ folder

### PR DESCRIPTION
**Pull request by KanclerzPiotr**
# Existing Sample Changes
## Description

Change occurrences of DPC++/DPCPP to SYCL inside the DirectProgramming/DPC++ folder. 

# NOTE That main DPC++ folder is not yet renamed.

Fixes Issue# 

## External Dependencies

Direct links to specific files can be broken

## Type of change

- [ ] Implement fixes for ONSAM Jiras

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Command Line

### CODE REVIEW

File: DirectProgramming/DPC++/DenseLinearAlgebra/matrix_mul/Makefile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 DPCPP_CXX = dpcpp
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/DenseLinearAlgebra/matrix_mul/Makefile:1]: Do we want to rename DPCPP to SYCL in the Makefile or maybe get rid of it altogether?

File: DirectProgramming/DPC++/Jupyter/.vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/Jupyter/.vscode/settings.json:1]: What's this file for? Did you accidentally check it in?

File: DirectProgramming/DPC++/DenseLinearAlgebra/matrix_mul/Makefile (hunk 1 above)
Comment (jimmytwei) [inline DirectProgramming/DPC++/DenseLinearAlgebra/matrix_mul/Makefile:1]: @KanclerzPiotr Do we want to rename DPCPP to SYCL in the Makefile or maybe get rid of it altogether?

File: DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/07_SYCL_Library/oneDPL_gamma_correction.ipynb [hunk 3]
```diff
@@ -278,6 +278,11 @@
    "toc_section_display": true,
    "toc_window_display": true
   },
+  "vscode": {
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/07_SYCL_Library/oneDPL_gamma_correction.ipynb:281]: @KanclerzPiotr This section wasn't in the original version. Remove?

File: DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/12_SYCL_Local_Memory_And_Atomics/run_localmem_info.sh [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 source /opt/intel/oneapi/setvars.sh > /dev/null 2>&1
-/bin/echo "##" $(whoami) is compiling DPCPP_Essentials Module12 -- DPCPP Atomics Local Memory - 3 of 5 localmem_info.cpp
+/bin/echo "##" $(whoami) is compiling SYCL_Essentials Module12 -- SYCL tomics Local Memory - 3 of 5 localmem_info.cpp
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/12_SYCL_Local_Memory_And_Atomics/run_localmem_info.sh:3]: Change "SYLC tomics" to "SYCL Atomics"

File: DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/12_SYCL_Local_Memory_And_Atomics/run_matrixmul_16x16_localmem.sh [hunk 5]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 source /opt/intel/oneapi/setvars.sh > /dev/null 2>&1
-/bin/echo "##" $(whoami) is compiling DPCPP_Essentials Module12 -- DPCPP Atomics Local Memory - 5 of 5 matrixmul_16x16_localmem.cpp
+/bin/echo "##" $(whoami) is compiling SYCL_Essentials Module12 -- SYCL tomics Local Memory - 5 of 5 matrixmul_16x16_localmem.cpp
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/12_SYCL_Local_Memory_And_Atomics/run_matrixmul_16x16_localmem.sh:3]: @KanclerzPiotr Change "SYLC tomics" to "SYCL Atomics".

File: DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/Makefile [hunk 6]
```diff
@@ -93,121 +93,121 @@ prog31: 07_DPCPP_Library/lab/maximum_function.cpp
...
+	02_SYCL_Program_Structure/bin/prog5
+	02_SYCL_Program_Structure/bin/prog6
+	03_SYCL_Unified_Shared_Memory/prog7
+	03_SYCL_Unified_Shared_Memory/prog8
+	03_SYCL_Unified_Shared_Memory/prog9
+	03_SYCL_Unified_Shared_Memory/prog10
+	04_SYCL_Sub_Groups/prog11
+	04_SYCL_Sub_Groups/prog12
+	04_SYCL_Sub_Groups/prog13
 	07_DPCPP_Library/bin/prog14
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/Makefile:211]: @KanclerzPiotr Should "07_DPCPP_Library" be renamed to "07_SYCL_Library"?

File: DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/Makefile [hunk 7]
```diff
@@ -227,39 +227,39 @@ Run:
...
...YCL_Buffers_And_Accessors_Indepth/bin/prog40 09_SYCL_Buffers_And_Accessors_Indepth/bin/prog41 09_SYCL_Buffers_And_Accessors_Indepth/bin/prog42 09_SYCL_Buffers_And_Accessors_Indepth/bin/prog43 09_SYCL_Buffers_And_Accessors_Indepth/bin/prog44 09_SYCL_Buffers_And_Accessors_Indepth/bin/prog45 09_SYCL_Buffers_And_Accessors_Indepth/bin/prog46 09_SYCL_Buffers_And_Accessors_Indepth/bin/prog47 09_SYCL_Buffers_And_Accessors_Indepth/bin/prog48 10_SYCL_Graphs_Scheduling_Data_management/bin/prog49 10_SYCL_Graphs_Scheduling_Data_management/bin/prog50 10_SYCL_Graphs_Scheduling_Data_management/bin/prog51 10_SYCL_Graphs_Scheduling_Data_management/bin/prog52 10_SYCL_Graphs_Scheduling_Data_management/bin/prog53 10_SYCL_Graphs_Scheduling_Data_management/bin/prog54 10_SYCL_Graphs_Scheduling_Data_management/bin/prog55 10_SYCL_Graphs_Scheduling_Data_management/bin/prog56 04_SYCL_Sub_Groups/prog57 04_SYCL_Sub_Groups/prog58 04_SYCL_Sub_Groups/prog59 11_Intel_Distribution_for_GDB/prog60 08_SYCL_Reduction/prog61 12_SYCL_Local_Memory_And_Atomics/prog62 12_SYCL_Local_Memory_And_Atomics/prog63 12_SYCL_Local_Memory_And_Atomics/prog64 12_SYCL_Local_Memory_And_Atomics/prog65 12_SYCL_Local_Memory_And_Atomics/prog66
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/Makefile:264]: @KanclerzPiotr Should "07_DPCPP_Library" be renamed to "07_SYCL_Library"?

File: DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/README.md [hunk 8]
```diff
@@ -38,13 +38,13 @@ Third party program Licenses can be found here: [third-party-programs.txt](https
...
+|[SYCL Buffers and Accessors in depth](09_SYCL_Buffers_And_Accessors_Indepth/SYCL_Buffers_accessors.ipynb)| + Buffers and Accessors<br>+ Buffer properties and usecases<br>+ Create Sub-buffers<br>+ Host accessors and usecases<br>+ ___Lab Exercise___: Buffers and Accessors
+|[SYCL Task Scheduling and Data Dependency](10_SYCL_Graphs_Scheduling_Data_management/SYCL_Task_Scheduling_Data_dependency.ipynb)| + Different types of data dependencies<br>+ Execution of graph scheduling<br>+ modes of dependencies in Graphs scheduling<br>+ ___Lab Exercise___: Task Scheduling | 120 min|
+|[SYCL Local Memory And Atomics](12_SYCL_Local_Memory_And_Atomics/Local_Memory_Atomics.ipynb)| + Query Local memory type and size<br>+ Local memory and Group barriers<br>+ Local Accessor usage<br>+ Atomic Operations Buffers<br>+ Atomic Operations USM<br>+ ___Lab Exercise___: Atomic Operation | 90 min|
 |[Intel® oneAPI DPC++ Library (oneDPL)](07_DPCPP_Library/oneDPL_Introduction.ipynb)| + Introduction to Intel oneAPI DPC++ Library (oneDPL)<br>+ ___Lab Exercise___: Gamma Correction with oneDPL
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/README.md:48]: @KanclerzPiotr Should "07_DPCPP_Library" be renamed to "07_SYCL_Library"?

File: DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/TeacherKit.ipynb [hunk 9]
```diff
@@ -136,7 +136,12 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.9.10"
+   "version": "3.9.7"
+  },
+  "vscode": {
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/TeacherKit.ipynb:141]: @KanclerzPiotr Double check vscode section.

File: DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/TeacherKit.ipynb [hunk 10]
```diff
@@ -92,13 +92,13 @@
...
+    "|[SYCL Buffers and Accessors in depth](09_SYCL_Buffers_And_Accessors_Indepth/SYCL_Buffers_accessors.ipynb)| + Buffers and Accessors<br>+ Buffer properties and usecases<br>+ Create Sub-buffers<br>+ Host accessors and usecases<br>+ ___Lab Exercise___: Buffers and Accessors | 120 min |\n",
+    "|[SYCL Task Scheduling and Data Dependency](10_SYCL_Graphs_Scheduling_Data_management/SYCL_Task_Scheduling_Data_dependency.ipynb)| + Different types of data dependency handling<br>+ Execution of graph scheduling<br>+ modes of dependencies in Graphs scheduling<br>+ ___Lab Exercise___: Task Scheduling | 120 min |\n",
+    "|[SYCL Local Memory And Atomics](12_SYCL_Local_Memory_And_Atomics/Local_Memory_Atomics.ipynb)| + Query Local memory type and size<br>+ Local memory and Group barriers<br>+ Local Accessor usage<br>+ Atomic Operations Buffers<br>+ Atomic Operations USM<br>+ ___Lab Exercise___: Atomic Operation | 90 min|\n",
     "|[Intel® oneAPI DPC++ Library (oneDPL)](07_DPCPP_Library/oneDPL_Introduction.ipynb)| + Introduction to Intel oneAPI DPC++ Library (oneDPL)<br>+ ___Lab Exercise___: Gamma Correction with oneDPL | 120 min |\n",
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/TeacherKit.ipynb:102]: @KanclerzPiotr Should "07_DPCPP_Library" be renamed to "07_SYCL_Library"?

File: DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/Welcome.ipynb [hunk 11]
```diff
@@ -101,6 +101,11 @@
    },
    "toc_section_display": true,
    "toc_window_display": true
+  },
+  "vscode": {
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/Welcome.ipynb:105]: @KanclerzPiotr Check "vscode" section.

File: DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/Welcome.ipynb [hunk 12]
```diff
@@ -42,19 +42,19 @@
     "## Module 7 - [Intel® oneAPI DPC++ Library](07_DPCPP_Library/oneDPL_Introduction.ipynb)\n",
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/Jupyter/oneapi-essentials-training/Welcome.ipynb:42]: @KanclerzPiotr Should "07_DPCPP_Library" be renamed to "07_SYCL_Library"?

File: DirectProgramming/DPC++/Jupyter/sycl-performance-portability-training/00_Introduction.ipynb [hunk 13]
```diff
@@ -388,7 +388,12 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.9.10"
+   "version": "3.9.7"
+  },
+  "vscode": {
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/Jupyter/sycl-performance-portability-training/00_Introduction.ipynb:393]: @KanclerzPiotr Check "vscode" section.

File: DirectProgramming/DPC++/DenseLinearAlgebra/matrix_mul/Makefile.win [hunk 14]
```diff
@@ -1,20 +1,20 @@
 DPCPP_CXX = dpcpp
```
Comment (jimmytwei) [inline DirectProgramming/DPC++/DenseLinearAlgebra/matrix_mul/Makefile.win:1]: @KanclerzPiotr Do we want to rename DPCPP to SYCL in the Makefile or maybe get rid of it altogether?

**Comment by jimmytwei** [comment]
@KanclerzPiotr Please review my comments and make the changes if needed. Also, do the folders iso2dfd_dpcpp and iso3dfd_dpcpp need to be renamed?

Also, can you remove the ProjectTemplates folder (to resolve the above conflicts)? I approved and merged a PR to delete the entire templates folder.

**Comment by KanclerzPiotr** [comment]
dpcpp in cmake/make files will be changed under different PR with the compiler change.
I will remove the conflict.
About .vscode i will try to fix this.
I dont think we should change the DPCPP_Library since this is related to the oneDPL which is not part of the SYCL spec but only in the oneAPI

**Comment by KanclerzPiotr** [comment]
Im closing this PR because broke this with merge changes and rebase after this

**Pull request closed without merging**